web/assets/javascripts/base-charts.js

```diff
@@ -0,0 +1,104 @@
+if (window.matchMedia("(prefers-color-scheme: dark)").matches) {
+  Chart.defaults.borderColor = "#333";
+  Chart.defaults.color = "#aaa";
+}
+
+class Colors {
+  constructor() {
+    this.assignments = {};
+    this.success = "#006f68";
+    this.failure = "#af0014";
+    this.fallback = "#999";
+    this.primary = "#537bc4";
+    this.available = [
```

Comment:
I can't quite visualize what this will look like, but if I may suggest choosing colorblind-safe palettes for anything where colors must be distinguished, e.g. .https://lospec.com/palette-list/ibm-color-blind-safe